src/Microsoft.AspNetCore.Blazor.Razor.Extensions/ComponentDocumentRewritePass.cs

```diff
@@ -218,48 +217,55 @@ private void RewriteChildren(RazorSourceDocument source, IntermediateNode node)
                                             stack.Pop();
```

Comment:
So the underlying issue is that L215-219 were running for BOTH start tag and end tag, and they were really only meant for the start tag. I'm not sure why these were both the same case of the switch statement, so I undid that and my life got instantly better.

Reply:
![Great](https://i.pinimg.com/736x/c4/dd/a0/c4dda060f3bb6ab68bf024b03a924621--background-images-an-eye.jpg)